scompose/project/project.py

```diff
@@ -166,6 +168,23 @@ def parse(self):
                     working_dir=self.working_dir,
                 )
 
+            # Eventually reorder instances based on depends_on constraints
+            sorted_instances = []
+            for instance in self.instances:
+                depends_on = self.instances[instance].params.get("depends_on", [])
+
+                try:
+                    index = sorted_instances.index(instance)
```

Comment:
So this is to say that we parse depends on, and if we don't find the instance that it depends on, we add to the sorted_instances list anyway and the index is for that instance instead. Then finally we loop over once more, and put any missing ones at the end. This would mean that the ones without dependencies are removed last - that makes sense.

I think since this is adding a large chunk of code to the initial parse function, and since it could be useful to have this other times to get a list of sorted instances based on dependencies, this would do well to be a separate function called here. It should take in some current set of instances, and return the sorted set. It shouldn't be setting or changing anything for self.instances - this should be done for the calling function. That way, we can also write a test for it.

Reply:
> this should be done for the calling function. That way, we can also write a test for it.

Makes total sense, I am gonna do it!

Reply:
Would you see it better as Project's method (as in f864a1d) or as a standalone method in utils?

Reply:
Let's think about this - so the project instance is meant to load and represent a project file (singularity-compose.yml) and doesn't even load without it. Currently we use utils for functions to read / write / otherwise interact with yaml or files, so my thinking is that I don't see any obvious use cases where we would want to have the function outside of the project file - using the Project class always helps the user by handling the load and some parsing, so my thinking is that it should be a project method. What are your thoughts? Are there use cases for having it external that I'm not thinking of?

Reply:
Yeah, it makes sense to keep as Project's method, I've also indirectly tested it in the new commit.